skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java

```diff
@@ -5,14 +5,24 @@
 import com.RDS.skilltree.Skill.SkillRepository;
 import com.RDS.skilltree.User.UserModel;
 import com.RDS.skilltree.User.UserRepository;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import jakarta.persistence.EntityNotFoundException;
 import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.ClassPathResource;
 import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
 import org.springframework.data.domain.PageRequest;
 import org.springframework.stereotype.Service;
 
+import java.io.IOException;
+import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Service
 @RequiredArgsConstructor
```

Comment:
why EndorsementService is not an Interface..

Reply:
I see EndorsementService is an interface and EndorsementServiceImpl implements it. Is that what you were asking? This was an already existing structure .. let me know if any suggestions

Reply:
which

> I see EndorsementService is an interface and EndorsementServiceImpl implements it. Is that what you were asking? This was an already existing structure .. let me know if any suggestions

which design pattern you are using here, answer this first

Reply:
I think it is a form of dependency injection where EndorsementServiceImpl implements EndorsementService interface ... and in EndorsementController Spring boot injects an instance of EndorsementServiceImpl into the constructor ..
I might be wrong , could you help me understand what is being used here?